crates/router/src/connector/nmi/transformers.rs

```diff
@@ -71,6 +77,292 @@ impl<T>
+                            errors::ConnectorError::InvalidConnectorConfig {
+                                config: "public_key",
+                            },
+                        )?,
+                    }),
+                    mandate_reference: None,
+                    connector_metadata: None,
+                    network_txn_id: None,
+                    connector_response_reference_id: None,
+                    incremental_authorization_allowed: None,
+                }),
+                enums::AttemptStatus::AuthenticationPending,
+            ),
+            Response::Declined | Response::Error => (
+                Err(types::ErrorResponse {
+                    code: item.response.response_code,
+                    message: item.response.responsetext,
+                    reason: None,
+                    status_code: item.http_code,
+                    attempt_status: None,
+                    connector_transaction_id: None,
+                }),
+                enums::AttemptStatus::Failure,
+            ),
+        };
+        Ok(Self {
+            status,
+            response,
+            ..item.data
+        })
+    }
+}
+
+#[derive(Debug, Serialize)]
+pub struct NmiCompleteRequest {
+    #[serde(rename = "type")]
+    transaction_type: TransactionType,
+    security_key: Secret<String>,
+    cardholder_auth: CardHolderAuthType,
+    cavv: String,
+    xid: String,
+    three_ds_version: ThreeDsVersion,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum CardHolderAuthType {
+    Verified,
+    Attempted,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub enum ThreeDsVersion {
+    #[serde(rename = "2.0.0")]
+    VersionTwo,
+    #[serde(rename = "2.2.0")]
+    VersionTwoPointTwo,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NmiRedirectResponseData {
+    cavv: String,
+    xid: String,
+    card_holder_auth: CardHolderAuthType,
+    three_ds_version: ThreeDsVersion,
+}
+
+impl TryFrom<&NmiRouterData<&types::PaymentsCompleteAuthorizeRouterData>> for NmiCompleteRequest {
+    type Error = Error;
+    fn try_from(
+        item: &NmiRouterData<&types::PaymentsCompleteAuthorizeRouterData>,
+    ) -> Result<Self, Self::Error> {
+        let transaction_type = match item.router_data.request.is_auto_capture()? {
+            true => TransactionType::Sale,
+            false => TransactionType::Auth,
+        };
+        let auth_type: NmiAuthType = (&item.router_data.connector_auth_type).try_into()?;
+        let payload_data = item
+            .router_data
+            .request
+            .get_redirect_response_payload()?
+            .expose();
+
+        let three_ds_data: NmiRedirectResponseData = serde_json::from_value(payload_data)
+            .into_report()
+            .change_context(errors::ConnectorError::MissingConnectorRedirectionPayload {
+                field_name: "cavv",
```

Comment:
change "cavv" to "three_ds_data" as cavv is one of the param inside three_ds_data